examples/samples/movingMessagesToDLQ.ts

```diff
@@ -0,0 +1,183 @@
+  const onMessageHandler: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    console.log(">>>>> Deadletter-ing the message", brokeredMessage);
+    await brokeredMessage.deadLetter();
+  };
+
+  const receiverHandler = await client.receive(onMessageHandler, onError, { autoComplete: false });
+  await delay(receiveClientTimeoutInMilliseconds);
+  await receiverHandler.stop();
+  await client.close();
+}
+
+async function runExceedMaxRetriesScenario(): Promise<void> {
+  // Prepare given queue by sending few messages
+  await sendMessage(1);
+
+  // Process messages from queue - exceed max retries on the messages
+  console.log("\n Setup queues, now running scenario B ... ");
+  const client = ns.createQueueClient(queuePath, { receiveMode: ReceiveMode.peekLock });
+  const onMessageHandler: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    console.log(
+      ">>>>> Attempt to abandon message. Current delivery count: ",
+      brokeredMessage.deliveryCount
+    );
+    await brokeredMessage.abandon();
+  };
+
+  const receiverHandler = await client.receive(onMessageHandler, onError, { autoComplete: false });
+  await delay(receiveClientTimeoutInMilliseconds);
+  await receiverHandler.stop();
+  await client.close();
+}
+
+async function runProcessingErrorHandlingScenario(): Promise<void> {
+  // Prepare given queue by sending few messages
+  await sendMessage(2);
+
+  // Process messages from queue, by invoking a faulty processor that doesn't successfully process
+  console.log("\n Setup queues, now running scenario C ... ");
+  const client = ns.createQueueClient(queuePath, { receiveMode: ReceiveMode.peekLock });
+  const onMessageHandler: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    console.log(">>>>> Rejecting the message: ", brokeredMessage);
+    try {
+      throw new Error("Customer is Vegetarian.");
+    } catch (err) {
+      console.log("Message from faulty processor: ", err);
+
+      // TODO: Fix bug with .deadletter() in SDK for not setting these properties on the message.
+      await brokeredMessage.deadLetter({
+        deadletterReason: "Incorrect Recipe type",
+        deadLetterErrorDescription: "Recipe type does not  match preferences."
+      });
+    }
+  };
+
+  const receiverHandler = await client.receive(onMessageHandler, onError, {
+    autoComplete: false
+  });
+  await delay(receiveClientTimeoutInMilliseconds);
+  await receiverHandler.stop();
+  await client.close();
+}
+
+// OnError handler for message receive
+const onError: OnError = (err: MessagingError | Error) => {
+  console.log(">>>>> Error occurred: ", err);
+};
+
+// Helper to send few sample messages to given queue
+async function sendMessage(index: number): Promise<void> {
+  const nonVegetarianRecipes = [
+    { name: "Grilled Steak", type: "Non-Vegetarian" },
+    { name: "Dry-braised Schezuan Prawn", type: "Non-Vegetarian" },
+    { name: "Creamy Chicken Pasta", type: "Non-Vegetarian" }
```

Comment:
"Non-Vegetarian" is an [Indian English word](https://en.wikipedia.org/wiki/Non-vegetarian), we should go for a more global example :)